specification.markdown

```diff
@@ -228,6 +210,95 @@ VINT\_WIDTH | VINT\_MARKER | VINT\_DATA     | Element Data Size Status
         For example -2 can be coded as 0xFFFFFFFFFFFFFE or 0xFFFE or
         0xFE and 5 can be coded 0x000000000005 or 0x0005 or 0x05.
 
+## EBML Element Types
+
+Each defined EBML Element MUST have a declared Element Type. The Element Type defines a concept for storing data that may be constrained by length, endianness, and purpose.
+
+Element Data Type:   Signed Integer
+
+    Endianness:     Big-endian
+    Length:         A Signed Integer Element MUST at least 1 octet in
```

Comment:
Mh, looking at the code of libebml, when reading an integer the default value is set to 0. But I can't find in the logic if the element data is actually read when the size is 0 or not. It's possible the value is not actually read and when picking the value for the element, the default value is used.

@mbunkus can you confirm which way you use it ? In https://github.com/Matroska-Org/libebml/blob/master/src/EbmlMaster.cpp#L423 the `MaxSizeToRead` can be 0 and the element is Found, but not added to the `ElementList` which means it's lost. That seems like a bug to me.

I looked in `libebml2` and the there when the value of an integer is 0, the value 0 is used.

Also in the library there are checks in place so that an element which wasn't set and is not mandatory will crash. So an integer with a size of 0 will never be written by `libebml`, unless it's mandatory.